.github/workflows/e2e.yaml

```diff
@@ -2,44 +2,42 @@ name: End-to-End Tests
 on:
   workflow_dispatch:
   pull_request:
-concurrency: build-wsl
+concurrency: end-to-end-tests
+
 jobs:
-  start_vm:
+  start-vm:
     strategy:
       max-parallel: 1

```

Comment:
I really think we should eliminate this on the long run (let’s not block this PR on this) by:
- have a lock in the composite action itself to avoid multiple runs
- have the start VM idemnpotent
- maybe have a idling system on the VM to stop the machine? Then, we no longer rely on the last job to run.

We can then maybe further work in the composite action to eliminate the need for start-vm, have the first connect job doing it, and maybe make the second job depending on a particular step on the first one (this used to be possible in the early days of github action, but we need to see if this is still the case).

Reply:
Agreed on all fronts. The Action that 3rd parties will want to write will be similar to the one in `wsl-exec-example.yaml` which is much simpler than the one in `e2e.yaml`.